library/worldgen/biome/src/main/java/org/quiltmc/qsl/worldgen/biome/api/codec/ModifyEffectsModifier.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.worldgen.biome.api.codec;
+
+import java.util.Optional;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+import net.minecraft.registry.Holder;
+import net.minecraft.sound.BiomeAdditionsSound;
+import net.minecraft.sound.BiomeMoodSound;
+import net.minecraft.sound.MusicSound;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+import net.minecraft.world.biome.BiomeEffects;
+import net.minecraft.world.biome.BiomeParticleConfig;
+
+import org.quiltmc.qsl.data.callbacks.api.predicate.CodecAwarePredicate;
+import org.quiltmc.qsl.worldgen.biome.api.BiomeModificationContext;
+import org.quiltmc.qsl.worldgen.biome.api.BiomeModifier;
+import org.quiltmc.qsl.worldgen.biome.api.BiomeSelectionContext;
+
+/**
+ * A biome modifier that modifies the effects of a biome.
+ * @see BiomeModificationContext.EffectsContext
+ */
+public record ModifyEffectsModifier(CodecAwarePredicate<BiomeSelectionContext> selector,
+									Optional<Integer> fogColor,
```

Comment:
Can't we use OptionalInt for those?... and I swear if that's a DFU restriction.

Reply:
We probably could with an xmap, though I'm not sure of the advantage, honestly - you'd still be serializing or deserializing it as an `Optional<Integer>` because, well, DFU is all generic stuff; but yeah, that's easy enough to switch around

Reply:
Mmh, with DFU forcing an Optional<Integer> anyway there's not a big insensitive to switch then.